### PR TITLE
fix: add PATH-based Unix shell detection for stripped environments

### LINKED ISSUE #3366: [Bug]: non-interactive-env hook generates PowerShell $env: syntax on Git Bash (Windows)
### Prerequisites

- [x] I will write this issue in English (see our [Language Policy](https://github.com/code-yeongyu/oh-my-opencode/blob/dev/CONTRIBUTING.md#language-policy))
- [x] I have searched existing issues to avoid duplicates
- [x] I am using the latest version of oh-my-opencode
- [x] I have read the [documentation](https://github.com/code-yeongyu/oh-my-opencode#readme) or asked an AI coding agent with this project's GitHub URL loaded and couldn't find the answer

### Bug Description

The `non-interactive-env` hook generates PowerShell `$env:` syntax on Windows, but the bash tool runs Git Bash (a Unix shell). This causes every `$env:VAR=value` assignment to be interpreted as a command name, producing `command not found` errors on every bash tool invocation.

The root cause is in `src/shared/shell-env.ts`: `detectShellType()` returns `"powershell"` on Windows because `PSModulePath` is always set by the system — even when the active shell is Git Bash. The `SHELL` env var check (which would correctly return `"unix"`) is never triggered because `$SHELL` is not set in the OpenCode process environment.

### Steps to Reproduce

1. Install oh-my-openagent plugin on Windows 11
2. Install Git Bash as the default terminal
3. Open any OpenCode session (inside or outside any project)
4. Run any `bash` tool command (e.g., `git status`, `ls`, `cargo check`)
5. Observe stderr output prefixed with 15 `command not found` errors

### Expected Behavior

The `non-interactive-env` hook should detect Git Bash (or any Unix-compatible shell on Windows) and generate Unix-compatible `export VAR=value;` syntax.

### Actual Behavior

PowerShell `$env:VAR=value;` syntax is prepended to every bash command, producing stderr noise on every invocation:

```
/usr/bin/bash: line 1: :CI=true: command not found
/usr/bin/bash: line 1: :DEBIAN_FRONTEND=noninteractive: command not found
/usr/bin/bash: line 1: :GIT_TERMINAL_PROMPT=0: command not found
/usr/bin/bash: line 1: :GCM_INTERACTIVE=never: command not found
/usr/bin/bash: line 1: :HOMEBREW_NO_AUTO_UPDATE=1: command not found
/usr/bin/bash: line 1: :GIT_EDITOR=:: command not found
/usr/bin/bash: line 1: :EDITOR=:: command not found
/usr/bin/bash: line 1: :VISUAL=: command not found
/usr/bin/bash: line 1: :GIT_SEQUENCE_EDITOR=:: command not found
/usr/bin/bash: line 1: :GIT_MERGE_AUTOEDIT=no: command not found
/usr/bin/bash: line 1: :GIT_PAGER=cat: command not found
/usr/bin/bash: line 1: :PAGER=cat: command not found
/usr/bin/bash: line 1: :npm_config_yes=true: command not found
/usr/bin/bash: line 1: :PIP_NO_INPUT=1: command not found
/usr/bin/bash: line 1: :YARN_ENABLE_IMMUTABLE_INSTALLS=false: command not found
```

The actual command still runs (these are stderr-only, non-fatal), but it wastes context tokens and pollutes tool output.

### Doctor Output

```shell
N/A — not a doctor issue, this is a shell detection bug
```

### Error Logs

```shell
# Full stderr from a simple `git status` command:
/usr/bin/bash: line 1: :CI=true: command not found
/usr/bin/bash: line 1: :DEBIAN_FRONTEND=noninteractive: command not found
/usr/bin/bash: line 1: :GIT_TERMINAL_PROMPT=0: command not found
/usr/bin/bash: line 1: :GCM_INTERACTIVE=never: command not found
/usr/bin/bash: line 1: :HOMEBREW_NO_AUTO_UPDATE=1: command not found
/usr/bin/bash: line 1: :GIT_EDITOR=:: command not found
/usr/bin/bash: line 1: :EDITOR=:: command not found
/usr/bin/bash: line 1: :VISUAL=: command not found
/usr/bin/bash: line 1: :GIT_SEQUENCE_EDITOR=:: command not found
/usr/bin/bash: line 1: :GIT_MERGE_AUTOEDIT=no: command not found
/usr/bin/bash: line 1: :GIT_PAGER=cat: command not found
/usr/bin/bash: line 1: :PAGER=cat: command not found
/usr/bin/bash: line 1: :npm_config_yes=true: command not found
/usr/bin/bash: line 1: :PIP_NO_INPUT=1: command not found
/usr/bin/bash: line 1: :YARN_ENABLE_IMMUTABLE_INSTALLS=false: command not found
```

### Configuration

```json
{
  "$schema": "https://raw.githubusercontent.com/code-yeongyu/oh-my-openagent/dev/assets/oh-my-opencode.schema.json",
  "autoupdate": "notify",
  "permission": {
    "lsp": "allow"
  },
  "watcher": {
    "ignore": [
      "**/.git/**",
      "**/node_modules/**",
      "**/target/**"
    ]
  },
  "plugin": [
    "file:///C:/Users/zir/.config/opencode/plugin/oh-my-openagent/dist/index.js"
  ]
}
```

### Additional Context

System:
Windows 11 (with Git Bash / MSYS2)

**Bug location chain:**

| File | Line | Issue |
|---|---|---|
| `src/shared/shell-env.ts` | 24 | `detectShellType()` checks `PSModulePath` before platform fallback, always returns `"powershell"` on Windows |
| `src/hooks/non-interactive-env/non-interactive-env-hook.ts` | 55 | Uses `process.platform === "win32" ? "powershell" : "unix"` as fallback — correct for cmd, wrong for Git Bash |
| `src/hooks/non-interactive-env/non-interactive-env-hook.ts` | 56 | `buildEnvPrefix(env, "powershell")` generates `$env:VAR='val';` syntax |

**Suggested fix:** In `src/shared/shell-env.ts`, check for `process.env.TERM` (set by Git Bash/WSL, typically `"xterm-256color"`) or `process.env.BASH_VERSION` to detect Unix-compatible shells on Windows. Alternatively, in the hook at line 55, check for these env vars before defaulting to `"powershell"`.

**Workaround:** Set `SHELL=/usr/bin/bash` in the OpenCode process environment — the existing `detectShellType()` already prioritises `$SHELL` over `$PSModulePath`.

### Operating System

Windows

### OpenCode Version

1.4.3

### PR DESCRIPTION
**Pull request by Zireael**
## Summary

Adds a third detection layer for Unix shells on Windows that catches cases where all shell-related environment variables have been stripped by the parent process.

## Problem

PR #3370 fixed the primary issue where `PSModulePath` (always set on Windows) was checked before Unix shell indicators. However, there's an edge case where:

1. A parent process spawns bash with a clean/minimal environment
2. All shell indicators (`SHELL`, `BASH_VERSION`, `MSYSTEM`, `WSL_DISTRO_NAME`) are stripped
3. Only `PSModulePath` and `process.platform` remain

In this case, `detectShellType()` would still return `"powershell"`.

## Solution

Added `detectUnixPathInPATH()` function that checks PATH for Unix shell indicators:

```typescript
function detectUnixPathInPATH(): boolean {
  const path = (process.env.PATH || process.env.Path || "").toLowerCase()
  if (path.includes("\usr\bin") || path.includes("/usr/bin")) return true
  if (path.includes("\msys64\bin") || path.includes("/msys64/bin")) return true
  if (path.includes("\cygwin\bin") || path.includes("/cygwin/bin")) return true
  if (path.includes("\git\usr\bin") || path.includes("/git/usr/bin")) return true
  return false
}
```

This is scoped to `process.platform === "win32"` and runs as layer 3 (after env vars, before PSModulePath).

## Detection Priority (updated)

1. `SHELL` env var → Unix shell (explicit)
2. Windows indicators → `BASH_VERSION`, `MSYSTEM`, `WSL_DISTRO_NAME`  
3. **NEW**: PATH fallback → `detectUnixPathInPATH()`
4. `PSModulePath` → PowerShell
5. Platform fallback → cmd/unix

## Verification

- Analyzed all 54 PATH entries on development machine
- No false positives: generic `/bin` dirs (`.cargo/bin`, user directories) correctly ignored
- Only actual Unix shell paths detected: MSYS2, Git Bash, Cygwin

## Testing

PR #3370 introduced regression tests. This edge case fix adds:

1. `detectUnixPathInPATH()` unit tests with mocked PATH
2. Integration test for stripped environment scenario

Closes #3366

---
## Summary by cubic
Adds a PATH-based fallback to detect Unix shells on Windows when env vars are stripped, avoiding false PowerShell detection. Narrows PATH matches to Windows-specific Unix tool dirs to prevent false positives.

- **Bug Fixes**
  - Added `detectUnixPathInPATH()` to scan PATH for `\git\usr\bin`, `\msys64\usr\bin`, or `\cygwin` on `win32`; removed generic `/usr/bin`, `\git\bin`, and `\mingw{32,64}\` checks.
  - Updated detection order: `SHELL` → `MSYSTEM`/`WSL_DISTRO_NAME` → PATH fallback → `PSModulePath` → platform default; removed `BASH_VERSION`.

<sup>Written for commit 9dda92e722b61b9b08356ce86099b906c89e8f27. Summary will update on new commits. <a href="https://cubic.dev/pr/code-yeongyu/oh-my-openagent/pull/3372?utm_source=github">Review in cubic</a></sup>

### CODE REVIEW

**Comment by Zireael** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by code-yeongyu** [comment]
[sisyphus-bot]

PR sweep first-pass triage on `dev`.

- Author: `Zireael`
- Title: `fix: add PATH-based Unix shell detection for stripped environments`
- Closing issues: #3366
- Review decision: none yet
- Mergeable state: **CONFLICTING** (rebase needed)
- CI status: green / no failing checks

Needs rebase + review. Please rebase onto current `dev` first; then a review can land cleanly.

Assigning `code-yeongyu` so the maintainer can prioritize a focused review of this PR. If the linked issue above has been triaged, the verdict there is the authoritative direction for this change.

File: src/shared/shell-env.ts [hunk 1]
```diff
@@ -35,6 +47,20 @@ export function detectShellType(): ShellType {
...
 }
 
+function detectUnixPathInPATH(): boolean {
+  const path = (process.env.PATH || process.env.Path || "").toLowerCase()
+  // Check for Windows-specific Unix tool installation directories.
+  // These contain backslash separators (Windows-only) and known
+  // subdirectory names used by Git Bash, MSYS2, and Cygwin.
+  // Generic paths like /usr/bin are NOT checked — they exist on every
+  // Linux system and produce false positives when platform is mocked.
+  if (path.includes("\\git\\")) return true
```
[the remark this answers is not in the document]
Comment (Zireael) [reply src/shared/shell-env.ts]: | Pattern | Before (too broad) | After (targeted) |
|---------|-------------------|-------------------|
| Git | `\\git\\` — matches `C:\Program Files\Git\bin` which is in PATH for *every* shell with Git installed | `\\git\\usr\\bin` — only `C:\Program Files\Git\usr\bin` is added by Git Bash itself |
| MSYS2 | `\\msys64\\` — matches `\msys64\mingw64\bin` which may be on system PATH without active MSYS2 | `\\msys64\\usr\\bin` — only set by MSYS2 Unix shell |
| Cygwin | `\\cygwin` — kept as-is (Cygwin's `\cygwin64\bin` only contains Unix tools) | same |
| MinGW | `\\mingw64\\` / `\\mingw32\\` — compiler toolchains, not Unix shell indicators | **removed** |

**Comment by code-yeongyu** [comment]
[sisyphus-bot] Hi Zireael. 🙏 Thanks for the third-layer PATH-based Unix shell detection (the edge case where PSModulePath is set but BASH_VERSION / MSYSTEM / WSL_DISTRO_NAME were stripped). The diagnosis is correct, and the narrowed `detectUnixPathInPATH` (msys64 / cygwin / git-usr-bin / /usr/bin) is the right shape.

Picking this back up from the 5/16 triage. The PR shows CONFLICTING against `dev`. `src/shared/shell-env.ts` has been touched by other in-flight work (especially the `shellSingleQuote` helper). The actual fix you want still applies, but the rebase resolution needs to:

- Keep the existing `BASH_VERSION || MSYSTEM || WSL_DISTRO_NAME` early-return block on dev (from #3370).
- Add your new `if (process.platform === "win32" && detectUnixPathInPATH()) return "unix"` immediately after, before the `PSModulePath` check.
- Keep the new `shellSingleQuote` export that landed on dev (it's not in your branch, so the rebase will need to preserve it).
- Add your `detectUnixPathInPATH` helper at the bottom of the file.

Could you rebase onto current `dev`, resolve the shell-env.ts conflict that way, and force-push? Once CI is green I'll come back for a focused review. The single-file scope of the change makes the rebase relatively quick, but the test file (if any) may also need a touch.

Really appreciate the careful narrowing on `detectUnixPathInPATH`; explicit `/usr/bin` / `msys64/bin` / etc. is much safer than a broad PATH heuristic.

**Comment by code-yeongyu** [comment]
[sisyphus-bot] Thanks for the stripped-environment shell detection follow-up! The parent issue is closed and the repo has restructured, so this no longer applies as-is. Closing — if you can still reproduce the edge case on current dev, please reopen a fresh PR.
